docs/application/native/guides/text-input/multi-assistant.md

```diff
@@ -0,0 +1,447 @@
+# Multi-assistant
+
+
+Multi-assistant allows the user to install and use multiple voice assistants on the same machine.
+
+The Multi-assistant framework manages voice assistants that are built with Multi-assistant client library and activates one among them when either the wake word is detected or the voice key is pressed.
+
+The main features of the Multi-assistant API include:
+
+- Retrieving information
+
+  You can [get various information](#info) from the Multi-assistant framework:
+
+  - Client state
+
+    The client state is changed by functions and applied as a precondition for each API.
+
+  - Service state
+
+    The service state is changed when the user starts or stops interacting with the installed voice assistants.
+
+  - Active state
+
+    Active state indicates whether the current client application is expected to serve as an active voice assistant for the user or not. An active voice assistant is responsible for responding to the user utterance and providing appropriate feedback.
+
+  - Audio Data
+
+    When the voice assistant gets activated, it retrieves the user utterance audio data from the Multi-assistant framework in order to process user's voice command.
+
+- Updating result information
+
+  You can [update different types of result information](#result) to the Mutli-assistant framework:
+
+  - ASR result
+
+    To provide visual feedback while processing the voice command, text representation that are converted from audio data needs to be updated as frequently as possible before the final result is retrieved.
+
+  - Result
+
+    Represents the final response to the user's query, including the summarized text information that will be displayed on the Multi-assistant panel if the device supports it.
+
+  - Recognition Result
+
+    Specifies whether the processing was successful or not.
+
+- Sending requests
+
+  - Adjust background volume
+
+    When a voice assistant starts playing the voice feedback, it can request to adjust the background volume for making the voice feedback more audible.
+
+To use functionalities of the Multi-assistant framework:
+
+1. Set up Multi-assistant client library and [register callbacks](#callback).
+
+   The initialization allows the Multi-assistant framework to distinguish your application from any other applications also using the framework. The registered callbacks allow you to receive notifications about changes in the client state, service state and active state, and enables retrieving user utterance audio data from the framework.
+
+2. Prepare the Multi-assistant client library.
+
+   The preparation connects the client application to the Multi-assistant service to start exchanging messages with the Mutli-assistant framework and wakeup engines.
+
+3. Retrieve audio data when activated.
+
+   If the registered active state changed event callback function is called with the ACTIVE parameter, then you must ask for the user utterance data to process the query by your voice assistant.
```

Comment:
IMO, adding "then you must" is necessary as it makes it mandate otherwise it becomes an option. 

Please correc tme if I am wrong,

Reply:
In my thought, the sentence without "then you must" is still an imperative. It is anyway demanding the user to "ask for the user utterance data", in case "the callback function is called with the parameter".

Anyway "then you must" works as additional words to emphasize the meaning, keeping it is also acceptable. If you want, we can keep it.